stdlib/open-uri.rb

```diff
@@ -0,0 +1,515 @@
+# Copied from https://raw.githubusercontent.com/ruby/ruby/373babeaac8c3e663e1ded74a9f06ac94a671ed9/lib/open-uri.rb
+# frozen_string_literal: true
+#require 'uri'
+require 'stringio'
+require 'time'
+
+module Kernel
+  private
+  alias open_uri_original_open open # :nodoc:
+  class << self
+    alias open_uri_original_open open # :nodoc:
+  end
+
+  # Allows the opening of various resources including URIs.
+  #
+  # If the first argument responds to the 'open' method, 'open' is called on
+  # it with the rest of the arguments.
+  #
+  # If the first argument is a string that begins with xxx://, it is parsed by
+  # URI.parse.  If the parsed object responds to the 'open' method,
+  # 'open' is called on it with the rest of the arguments.
+  #
+  # Otherwise, the original Kernel#open is called.
+  #
+  # OpenURI::OpenRead#open provides URI::HTTP#open, URI::HTTPS#open and
+  # URI::FTP#open, Kernel#open.
+  #
+  # We can accept URIs and strings that begin with http://, https:// and
+  # ftp://. In these cases, the opened file object is extended by OpenURI::Meta.
+  def open(name, *rest, &block) # :doc:
+    if name.respond_to?(:to_str) && %r{\A[A-Za-z][A-Za-z0-9+\-\.]*://} =~ name
```

Comment:
The original code was:
```ruby
 if name.respond_to?(:to_str) 
  && %r{\A[A-Za-z][A-Za-z0-9+\-\.]*://} =~ name 
  && (uri = URI.parse(name)).respond_to?(:open)
```

But `URI` is not (yet) implemented. Do you think we should implement URI first ?

Reply:
It's not that easy. Regular expressions in URI are very complex.

Reply:
Indeed, I have nightmares about regular expressions :rofl: